pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1008,14 +1018,18 @@ async def on_package(pkg):
                 Args:
                     pkg (createrepo_c.Package): A completed createrepo_c package.
                 """
+                if self.mirror and pkg.location_base:
+                    raise ValueError(XML_BASE_AND_MIRROR_INCOMPATIBLE_ERR_MSG)
```

Comment:
Will we ever get here? I would think that we'll stop when we see the location base in repomd.xml, or are you trying to cover all bases, when some parts of metadata have location base and some do not :/?

Reply:
I'm not sure if they can be configured separately or not, but it feels like it might just be an implementation detail even if so.  I'm not sure we should rely on it.

What a terrible feature, honestly.  I cannot for the life of me understand why you would want this level of granularity in sourcing packages.  I can *maybe* understand being able to declare separate roots for the packages and metadata, but there's a big difference between doing that and having (potentially) tens of thousands of individual base urls or repeating the same one tens of thousands of times.